serverless/packages/functions/src/trainspace/create_trainspace.ts

```diff
@@ -0,0 +1,84 @@
+import { APIGatewayProxyHandlerV2 } from "aws-lambda";
+import parseJwt from "@dlp-sst-app/core/parseJwt";
+import TrainspaceData from './trainspace-data';
+import { v4 as uuidv4 } from 'uuid';
+import { DynamoDBClient, PutItemCommand } from '@aws-sdk/client-dynamodb'; //@aws-sdk/client-dynamodb
+import { DynamoDBDocumentClient } from '@aws-sdk/lib-dynamodb';
+
+function validateRequestBody(eventBody: any, trainspace_id: string, uid: string) : TrainspaceData | null {
+    console.log(eventBody);
+    
+    switch(eventBody['data_source'])
+    {
+        case "TABULAR":
+            return new TrainspaceData(trainspace_id, uid, "TABULAR", eventBody['default'], eventBody['name'], 
+                { 
+                    criterion: eventBody['criterion'],
+                    optimizer_name: eventBody['optimizer_name'],
+                    shuffle: eventBody['shuffle'],
+                    epochs: eventBody['epochs'],
+                    batch_size: eventBody['batch_size'],
+                    user_arch: eventBody['user_arch']
+                }, "");
+        case "PRETRAINED":
+
+            return null;
+        case "IMAGE":
+            return new TrainspaceData(trainspace_id, uid, "IMAGE", eventBody['dataset_data'], eventBody['name'], eventBody['parameters_data'], eventBody['review_data']['notification_email']);
+        case "AUDIO":
+
+            return null;
+        case "TEXTUAL":
+
+            return null;
+        case "CLASSICAL_ML":
+
+            return null;
+        case "OBJECT_DETECTION":
+
+            return null;
+        default:
+            return null;
+    }
+}

```

Comment:
potential code smell. Anyway we can improve this code @alantao912 @farisdurrani @dwu359

Reply:
Should I have used polymorphism here instead ? I thought about it but I don't think that would get around having to check the string

Reply:
lets keep this for now but might wanna refactor later